uat/custom-components/src/main/java/com/aws/greengrass/artifacts/logGenerator.java

```diff
@@ -5,16 +5,200 @@
+        logFileExtension = args[1];
+        fileSize = Double.parseDouble(args[2]);
+        fileSizeUnit = args[3];
+        logWriteFreqSeconds = Long.parseLong(args[4]);
+        totalLogNumbers = Integer.parseInt(args[5]);
+
+        fileSizeBytes = getFileSizeInBytes(fileSize, fileSizeUnit);
+
+        String currentPath = System.getProperty("user.dir");
+
+        // configure the logger by setting the context, rolling policy, encoder
+        Logger logger = configureLogger(currentPath, logFileName, logFileExtension, fileSizeBytes);
+
+        // starting logging message
+        performLogging(logger, totalLogNumbers, logWriteFreqSeconds);
+
+        // print list of log files and results would be logged by GG nucleus, thus it would be eventually presented in
+        // local machine. This just provide an easier way to verify some logging info manually, and will not be used
+        // for UAT.
+        verifyLoggingSucceed(currentPath, logFileName, logFileExtension);
+    }
+
+    private Logger configureLogger(String currentPath, String fileName, String extension, int fileSizeBytes) {
+        LoggerContext loggerContext = new LoggerContext();
+        Logger logger = loggerContext.getLogger("logGenerator");
+
+        // appender: output destination
+        RollingFileAppender<ILoggingEvent> appender = new RollingFileAppender<>();
+        appender.setFile(currentPath + "/" + fileName + "." + extension);
+        appender.setAppend(true);
+        appender.setContext(loggerContext);
+
+        // rolling policy
+        SizeAndTimeBasedRollingPolicy rollingPolicy = new SizeAndTimeBasedRollingPolicy<>();
+        rollingPolicy.setMaxFileSize(new FileSize(fileSizeBytes));
+        rollingPolicy.setMaxHistory(MAX_HISTORY);
+        rollingPolicy.setFileNamePattern(currentPath + "/" + fileName + rotationNamePattern + "." + extension);
+        rollingPolicy.setParent(appender);
+        rollingPolicy.setContext(loggerContext);
+        appender.setRollingPolicy(rollingPolicy);
+        rollingPolicy.start();
+
+        // encoder
+        PatternLayoutEncoder encoder = new PatternLayoutEncoder();
+        encoder.setPattern("%d{yyyy-MM-dd HH:mm:ss} %-5level %logger{36} - %msg%n");
+        encoder.setContext(loggerContext);
+        encoder.start();
+
+        // attach encoder to appender, then start
+        appender.setEncoder(encoder);
+        appender.start();
+
+        // set logger
+        logger.addAppender(appender);
+        logger.setLevel(Level.INFO);
+
+        return logger;
+    }
+
+    private static void performLogging(Logger logger, int totalLogNumbers, long logWriteFreqSeconds) {
+        for (int i=0; i < totalLogNumbers; i++) {
+            random = Math.random();
+            StringBuilder sb = generateLogMessage(i, logger.getName());
+            logger.info(sb.toString());
+            try {
+                TimeUnit.SECONDS.sleep(logWriteFreqSeconds);
+            } catch (InterruptedException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)

Reply:
Same comment I had left here, call `System.exit(0)` if interrupted